component-catalog-connectors/connector-template/todo_catalog_connector/todo_catalog_connector.py

```diff
@@ -0,0 +1,128 @@
+
+        self.log.debug(f'Value of required parameter \'todo_required_parm\': {todo_required_parm}')
+        self.log.debug(f'Value of optional parameter \'todo_optional_parm\': {todo_optional_parm}')
+
+        try:
+            # connect to catalog and retrieve component list
+            self.log.debug('Retrieving component list from <TODO> catalog.')
+
+            # Fetch components and add their keys to the component list.
+            # Each 'todo_component_key_info' uniquely identifies a component in the catalog.
+            # Method read_catalog_entry has access to todo_component_key_info via the
+            # 'catalog_entry_data' parameter and uses it to retrieve the component from the catalog.
+            todo_component_key_info = {
+                'TODO-COMPONENT-KEY': 'dummy-component.yaml'
+                # ... compound keys are okay!
+            }
+            component_list.append(todo_component_key_info)
+
+        except Exception as ex:
+            # error handling
+            self.log.error(f'Error retrieving component list from <TODO> catalog: {ex}')
+
+        return component_list
+
+    def read_catalog_entry(self,
+                           catalog_entry_data: Dict[str, Any],
+                           catalog_metadata: Dict[str, Any]) -> Optional[str]:
+        """
+        Fetch the component that is identified by catalog_entry_data from
+        the <TODO> catalog.
+
+        :param catalog_entry_data: a dictionary that contains the information needed to read the content
+                                   of the component definition
+        :param catalog_metadata: the metadata associated with the catalog in which this catalog entry is
+                                 stored; in addition to catalog_entry_data, catalog_metadata may also be
+                                 needed to read the component definition for certain types of catalogs
+
+        :returns: the content of the given catalog entry's definition in string form
+        """
+
+        # Load user-provided catalog connector parameters.
+        todo_required_parm = catalog_metadata['todo_required_parm']
+        todo_optional_parm = catalog_metadata.get('todo_optional_parm')
+
+        # Get catalog component key information from the catalog_entry_data
+        # dictionary.
+        todo_component_key = catalog_entry_data.get('TODO-COMPONENT-KEY')
+
+        self.log.debug(f'Value of required parameter \'todo_required_parm\': {todo_required_parm}')
+        self.log.debug(f'Value of optional parameter \'todo_optional_parm\': {todo_optional_parm}')
+
+        # verify that a component key was provided
+        if todo_component_key is None:
+            # Log error and return None
+            self.log.error('Cannot connect to <TODO> catalog: todo_component_key must be provided.')
+            return None
+
+        # TODO: Load component from catalog using the provided key
```

Comment:
To reduce the number of `TODO`s in this file 😄 

```suggestion
        # load component from catalog using the provided key
```

Reply:
Well, it's a todo but fine (: